### DISCUSSION TITLE
Behavior and keybindings for signal list

### DISCUSSION THREAD

**Discussion by anihm136** [pesos/grofer#115] [General]
We now have a basic working implementation of a signal list (#112) to dispatch signals to processes. Now I'd like to discuss some of the finer points of how it works, and some useful keybindings to use it. In terms of behavior, this is how it currently works -
- Pressing the kill key (`K`/`<F9>`) opens the signal list (and, in essence, enters a kill submode)
- `j/k/<Up>/<Down>` allow navigation through the signal list
- `<Enter>` sends the highlighted signal to the selected process
- `K/<F9>` sends `SIGTERM` to the process (this is a slight change in behavior from before, see [this](https://www.man7.org/linux/man-pages/man7/signal.7.html), [this](https://stackoverflow.com/questions/5399355/difference-between-sigkill-sigterm-considering-process-tree) and [this](https://linuxhandbook.com/sigterm-vs-sigkill/) - bonus comic!) to evaluate the difference

What is sorely lacking is keybindings to navigate through the signal list. I find the keybindings used by `htop` (based on numbers) easy to use, and it should be fairly straightforward to implement. Additionally, we could discuss bringing in some of the other vim-like bindings used in other parts of the application if necessary.

Some of the behavioral aspects to discuss are -
- Ordering: Should we order the signals in the list in some specific way? Same as `htop`? Most used first? Alphabetical?
- Set of signals provided - The `syscall` library provides 35 signals as named constants. However, I observed only 31 signals supported by `kill` (using `kill -l`). This might be platform specific, or a limitation of `kill`. By contrast, `htop` provides 63 signals
- Safety - should we have an additional key (`<Enter>`/repeat key) to confirm? (this would now be in addition to two keys already)
- Exiting - `htop` does not allow exiting when in the kill submode, while the current implementation does. Personally, I feel it is fine (and even a bonus) since it does not cause any unexpected behavior

One last point of discussion is the UI. We currently only display the name of the signal. Additionally, we could show the integer representation of the signal, the numeric keybinding to select that signal (if we implement numeric navigation) and a short description of what the signal does.

**Comment by Gituser143** [comment]
- For ordering, I'm more inclined to sort either alphabetically, or on the integer codes. i say this because, it would be easy to implement a table sorting functionality by using existing functions :P
- For the set of signals, I'm out of my depth here, but for now the named constants should suffice I feel. We could extend the list once we figure out how.
- Safety is perfect as it is right now! `K` to open table and `<Enter>` to confirm signal. Anything more would be repetitive I feel.
- I agree with you on the exit. To be able to close even in the kill submode seems like a good choice!
- As for the integers for signals, I'm all for it! Seems like a good idea!

**Comment by MadhavJivrajani** [comment]
> Additionally, we could discuss bringing in some of the other vim-like bindings used in other parts of the application if necessary.

This is a great idea. Feel free to open an issue/start a discussion thread for deciding on adding vim-like keybindings. 

For the rest of the points, +1 to what @Gituser143 said.
